numpy/ma/core.py

```diff
@@ -6206,16 +6227,23 @@ def __str__(self):
         return str(masked_print_option._display)
 
     def __repr__(self):
-        return 'masked'
-
-    def flatten(self):
-        return masked_array([self._data], dtype=float, mask=[True])
+        if self is masked:
+            return 'masked'
+        else:
+            # something is wrong, make it obvious
+            return object.__repr__(self)
 
     def __reduce__(self):
         """Override of MaskedArray's __reduce__.
         """
         return (self.__class__, ())
 
+    # inplace operations have no effect. We have to override them to avoid
+    # trying to modify the readonly data and mask arrays
+    def __inop(self, other):
```

Comment:
Guess I named this to look like `__iadd__` without actually looking like a special method